### ISSUE TITLE
setting healthcheck disable: true in compose file prevents the service name from being resolvable

### ISSUE THREAD

**Issue by thecrazymonkey** [moby/moby#31341]
**Description**
I have very simple Dockerfile and docker-compose.yml files:

_Dockerfile_
```
FROM centos
HEALTHCHECK --interval=10s CMD exit 0
ENTRYPOINT while true; do tail -f /dev/null;done
```

_docker-compose.yml_
```
version: '3.1'

networks:
  default:
    driver: overlay
    ipam:
      driver: default
      config:
        - subnet: 192.168.24.0/24

services:
  health:
    image: healthbug 
    deploy:
      replicas: 1
    networks:
      - default 
    healthcheck:
      disable: true
```

**Steps to reproduce the issue:**
1. Build image from the Dockerfile above - "docker build -t healthbug ."
2. Run "docker stack deploy -c docker-compose.yml test"
3. "docker ps" should show the container as running without health status (expected)
4. Try "docker exec -ti <container_name> ping health"

**Describe the results you received:**
Ping returns
ping: health: Name or service not known

**Describe the results you expected:**
Expecting to see output of ping command - e..g.:
PING health (192.168.24.3) 56(84) bytes of data.
64 bytes from e0bdd9aab838 (192.168.24.3): icmp_seq=1 ttl=64 time=0.030 ms

**Additional information you deem important (e.g. issue happens only occasionally):**
For the failed scenario "docker inspect" command shows State as:
```
"State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 14956,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2017-02-24T17:17:35.368304641Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
```

If I comment out (or replace with test: exit 0) the healthcheck section from the docker compose file everything runs as expected and the container is reported as healthy after 10s interval. Service name resolves correctly.

**Output of `docker version`:**

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:50:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:50:14 2017
 OS/Arch:      linux/amd64
 Experimental: true

```

**Output of `docker info`:**

```
Containers: 7
 Running: 0
 Paused: 0
 Stopped: 7
Images: 588
Server Version: 1.13.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 499
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: s5ll7v7orfzxg37k2w22wlnl7
 Is Manager: true
 ClusterID: qwg1m234fmqm4ria30gtig3jg
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.1.2.30
 Manager Addresses:
  10.1.2.30:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-62-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 7.673 GiB
Name: devbox
ID: GMRV:JAVH:HVHY:SSJE:OJGA:KAFD:Z5JI:Q767:FTXM:GXAH:4NXN:ETAM
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 10.0.0.0/16
 172.26.0.0/16
 127.0.0.0/8
Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
This is on physical hw running either Ubuntu 16.04.2 or 14.04

**Comment by cpuguy83** [comment]
ping @sanimej

**Comment by mavenugo** [comment]
This seems related to the health-aware based SD. ping @dongluochen

**Comment by thaJeztah** [comment]
Also related https://github.com/docker/docker/issues/31342

**Comment by javierprovecho** [comment]
@thaJeztah #31342 was resolved/closed and the example above works fine on `17.06.0-ce`, you can close this.

**Comment by thaJeztah** [comment]
thanks! closing
